### DISCUSSION TITLE
Naming convention for action-area objects on pages

### DISCUSSION THREAD

**Discussion by Arthurvdv** [microsoft/alguidelines#223] [General]
With the introduction of the [Modern Action Bar](https://learn.microsoft.com/en-us/dynamics365-release-plan/2022wave2/smb/dynamics365-business-central/modern-action-bar) honestly I'm struggling a bit to have a consistent name convention these objects.

What could be the _best pratices_ on naming these?
* group
* action
* actionref

Example
```AL
actions
{
    area(processing)
    {
        group(Post) // PostGroup? PostGrp? Post_Group?
        {
            Caption = 'Post';

            action(Post)  // PostAction? PostAct? Post_Action?
            {
                Caption = 'Post';

                trigger OnAction()
                begin
                    Rec.Post();
                end;
            }
        }
    area(Promoted)
    {
        group(Post)  // PostPromotedGroup? PostPrmtdGrp? Post_Promoted_Group?
        {
            Caption = 'Post';

            actionref(Post; Post) { }  // PostPromoted? PostPrmtd? Post_Promoted? or PostActionref? PostActRef? Post_Actionref?
        }
    }
}
```

**Comment by NKarolak** [comment]
About promoted action **group** names and captions - the docs say it's either `Posting` or `Post/Print`, not Post:
https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/devenv-common-promoted-action-groups

Looking at the current standard:
action group Posting:
https://github.com/StefanMaron/MSDyn365BC.Code.History/blob/5295a85462ce6bbe739628a3c185591d87ab9582/BaseApp/Source/Base%20Application/SalesReceivables/Document/SalesOrder.Page.al#LL1941C19-L1941C19

The promoted area:
https://github.com/StefanMaron/MSDyn365BC.Code.History/blob/5295a85462ce6bbe739628a3c185591d87ab9582/BaseApp/Source/Base%20Application/SalesReceivables/Document/SalesOrder.Page.al#L2233

I'd always look for reference there (but without the ampersand in the captions).
However, good point - it's always quite a struggle, I totally agree.

**Comment by Arthurvdv** [comment]
_There are only two hard things in Computer Science: cache invalidation and naming things.
— Phil Karlton_

Luckily in AL we don't have to worry about cache invalidation ;-)

I've read the documentation and looked at some examples. I guess the thread running through the story is something like below?

```AL
actions
{
    area(processing)
    {
        group(Posting) // Plural of the action(s) name (and not explicitly specifying it's a group anchor)
        {
            Caption = 'Posting';

            action(Post)  // Singular name of the action
            {
                Caption = 'Post';

                trigger OnAction()
                begin
                    Rec.Post();
                end;
            }
        }
    area(Promoted)
    {
        group(Category_Post)  // Category_<GroupName>
        {
            Caption = 'Post';

            actionref(Post_Promoted; Post) { }  // <ActionName>_Promoted
        }
    }
}
```

It feels off to have a suffix on the Promoted area and nothing in the area's like processing. And why is a groupname in the Promoted area singular instead of plural?

Or do I need to let this go and can every developer decide for itself what they thinks it's best?
